src/utils/configUtils.ts

```diff
@@ -40,7 +41,25 @@ export async function loadRunConfig(workspaceFolder: WorkspaceFolder): Promise<I
             return defaultConfigs[0];
         }
     }
-    return await selectQuickPick(configItems, workspaceFolder);
+
+    const testItemMatch: RegExp = /testItem ~= \/(?<pattern>.*)\//;
```

Comment:
How about:

```typescript
    ...
    const candidateConfigItems: IExecutionConfig[] = filterCandidateConfigItems(configItems);
    return await selectQuickPick(candidateConfigItems, workspaceFolder);
}

function filterCandidateConfigItems(configItems: IExecutionConfig[]): IExecutionConfig[] {
    const matchPattern = /(?<contextKey>.+)\s*~=\s*\/(?<pattern>.+)\//;
    return configItems.filter((config: IExecutionConfig) => {
        const whenClause = config.when?.trim();
        if (whenClause) {
            const groups = matchPattern.exec(whenClause)?.groups;
            if (!groups) {
                return true;
            }
            switch (groups.contextKey) {
                case "testItem":
                    return testItemFiltering(testItems, groups.pattern);
                    break;
                default:
                    return true;
            }
        }
        return true;
    });
}

function testItemFiltering(testItems: TestItem[], pattern: string): boolean {
    ...
}

```

If we want to add some unit test cases, we can consider export checkTestItems() and filterCandidateConfigItems()

Reply:
https://github.com/microsoft/vscode-java-test/pull/1638/commits/469518512bb286d39bffa91279fab306fcca6c58